src/gwt/src/org/rstudio/studio/client/workbench/views/files/model/FileUploadToken.java

```diff
@@ -21,4 +21,24 @@
    protected FileUploadToken()
    {
    }
+
+   public final native String getFilename() /*-{
+      return this.filename;
+   }-*/;
+
+   public final native String getUploadTempFile() /*-{
+      return this.uploadedTempFile;
+   }-*/;
+
+   public final native String geTargetDirectory() /*-{
+      return this.targetDirectory;
+   }-*/;
+
+   public final native Boolean getUnzipFound() /*-{
```

Comment:
A small nit (not needed to change in this PR): GWT's JSNI is usually able to handle conversion to "native" Java types; e.g. you could use a plain `boolean` here to make the code (in my subjective opinion) cleaner below.